components-core/src/main/java/com/adyen/checkout/components/base/BaseConfigurationBuilder.kt

```diff
@@ -63,6 +66,18 @@ abstract class BaseConfigurationBuilder<
         return this as BuilderT
     }
 
+    /**
+     * TODO docs
+     */
+    fun setAmount(amount: Amount): BuilderT {
+        if (!CheckoutCurrency.isSupported(amount.currency) || amount.value < 0) {
+            throw CheckoutException("Currency is not valid.")
```

Comment:
Should we be more specific in this message? Cause either the currency or the value is invalid

Reply:
Maybe we should move the validation to the `Amount` class even?

Reply:
We can move it to `init` block of `Amount` class but this class is being used in response models too. We would be throwing exceptions while deserializing if backend doesn't send us a valid object. I think it's better to do the check for validity only for the `Amount`s merchants provide.